### PR TITLE
Limit payment history in sync

### PR DESCRIPTION
**Pull request by cob16**
- @StevenLeighton21 - Limit to a year of historic transaction history 

- @cob16 - require env var 'ENABLE_TENANCY_SYNC = true' to run snc jobs 
  - Would like some sugestions where the FeachureToggle module should live (directory wise)

### CODE REVIEW

File: lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ def self.for_tenancy(universal_housing_client, tenancy_ref)
               ) t
               WHERE row = 1
             )
-            DECLARE @RemainingTransactions INT = (SELECT COUNT(*) FROM [dbo].[rtrans] WITH (NOLOCK) WHERE tag_ref = @TenancyRef)
+            DECLARE @RemainingTransactions INT = (SELECT COUNT(*) FROM [dbo].[rtrans] WITH (NOLOCK) WHERE tag_ref = @TenancyRef AND post_date > DATEADD(year,-1,GETDATE()))
```
Comment (madetech-jeffpinkham) [inline lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb]: Suggest: 
```suggestion
            DECLARE @RemainingTransactions INT = (SELECT COUNT(tag_ref) FROM [dbo].[rtrans] WITH (NOLOCK) WHERE tag_ref = @TenancyRef AND post_date > DATEADD(year,-1,GETDATE()))
```
As ```Count(*)``` isn't efficient.

File: lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@ def self.for_tenancy(universal_housing_client, tenancy_ref)
...
             DECLARE @ActiveAgreementsCount INT = (SELECT COUNT(*) FROM [dbo].[arag] WITH (NOLOCK) WHERE tag_ref = @TenancyRef AND arag_status = @ActiveArrearsAgreementStatus)
```
Comment (madetech-jeffpinkham) [inline lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb]:
```suggestion
            DECLARE @ActiveAgreementsCount INT = (SELECT COUNT(tag_ref) FROM [dbo].[arag] WITH (NOLOCK) WHERE tag_ref = @TenancyRef AND arag_status = @ActiveArrearsAgreementStatus)
```

File: lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb [hunk 3]
```diff
@@ -21,7 +21,7 @@ def self.for_tenancy(universal_housing_client, tenancy_ref)
...
             DECLARE @BreachedAgreementsCount INT = (SELECT COUNT(*) FROM [dbo].[arag] WITH (NOLOCK) WHERE tag_ref = @TenancyRef AND arag_status = @BreachedArrearsAgreementStatus)
```
Comment (madetech-jeffpinkham) [inline lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb]:
```suggestion
            DECLARE @BreachedAgreementsCount INT = (SELECT COUNT(tag_ref) FROM [dbo].[arag] WITH (NOLOCK) WHERE tag_ref = @TenancyRef AND arag_status = @BreachedArrearsAgreementStatus)
```

File: lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb [hunk 4]
```diff
@@ -21,7 +21,7 @@ def self.for_tenancy(universal_housing_client, tenancy_ref)
...
             DECLARE @NospsInLastYear INT = (SELECT COUNT(*) FROM araction WITH (NOLOCK) WHERE tag_ref = @TenancyRef AND action_code = @NospActionDiaryCode AND action_date >= CONVERT(date, DATEADD(year, -1, GETDATE())))
```
Comment (madetech-jeffpinkham) [inline lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb]:
```suggestion
            DECLARE @NospsInLastYear INT = (SELECT COUNT(tag_ref) FROM araction WITH (NOLOCK) WHERE tag_ref = @TenancyRef AND action_code = @NospActionDiaryCode AND action_date >= CONVERT(date, DATEADD(year, -1, GETDATE())))
```

**Comment by madetech-jeffpinkham** [comment]
@cob16 @StevenLeighton21 made some suggestions on changing the select Count from ```*``` which selects every field on the table to select 1 field.

File: lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb (hunk 1 above)
Comment (StevenLeighton21) [reply lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb]: 🚢

Comment (StevenLeighton21) [reply lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb]: 🚢

Comment (StevenLeighton21) [reply lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb]: 🚢

Comment (StevenLeighton21) [reply lib/hackney/income/tenancy_prioritiser/universal_housing_criteria.rb]: 🚢

**Comment by StevenLeighton21** [comment]
Suggest we incorporate the `COUNT(tag_ref)` changes and merge?

**Pull request merged**